x/concentrated-liquidity/position.go

```diff
@@ -664,3 +665,105 @@ func (k Keeper) updateFullRangeLiquidityInPool(ctx sdk.Context, poolId uint64, l
 	osmoutils.MustSetDec(store, poolIdLiquidityKey, newTotalFullRangeLiquidity)
 	return nil
 }
+
+// transferPositions transfers ownership of a set of positions from a sender to a recipient.
+// It first checks if the provided position IDs are unique. If not, it returns a DuplicatePositionIdsError.
+// For each position ID, it retrieves the corresponding position and checks if the sender is the owner of the position.
+// If the sender is not the owner, it returns an error.
+// It then checks if the position has an active underlying lock, and if so, returns an error.
+// It then collects any outstanding incentives and rewards for the position, deletes the KVStore entries for the position,
+// and restores the position under the recipient's account.
+// If any of these operations fail, it returns the corresponding error.
+// If all operations succeed, it returns nil.
+func (k Keeper) transferPositions(ctx sdk.Context, positionIds []uint64, sender sdk.AccAddress, recipient sdk.AccAddress) error {
+	// All position IDs in the array must be unique.
+	if !osmoassert.Uint64ArrayValuesAreUnique(positionIds) {
+		return types.DuplicatePositionIdsError{PositionIds: positionIds}
+	}
+
+	// Fixed gas consumption per position ID to prevent spam
+	ctx.GasMeter().ConsumeGas(uint64(types.BaseGasFeeForTransferPosition*len(positionIds)), "cl transfer position fee")
```

Comment:
We should charge this gas prior to the uniqueness check above since that linearly iterates over all positions and can be used to halt the chain before we ever even get to the gas consumption step

Reply:
Resolving since you committed the fix for this